pkg/sdk/alert.go

```diff
@@ -0,0 +1,421 @@
+	return err
+}
+
+type DropAlertOptions struct {
+	drop  bool                   `ddl:"static" sql:"DROP"`  //lint:ignore U1000 This is used in the ddl tag
+	alert bool                   `ddl:"static" sql:"ALERT"` //lint:ignore U1000 This is used in the ddl tag
+	name  SchemaObjectIdentifier `ddl:"identifier"`
+}
+
+func (opts *DropAlertOptions) validate() error {
+	if !validObjectidentifier(opts.name) {
+		return ErrInvalidObjectIdentifier
+	}
+	return nil
+}
+
+func (v *alerts) Drop(ctx context.Context, id SchemaObjectIdentifier) error {
+	// alert drop does not support [IF EXISTS] so there are no drop options.
+	opts := &DropAlertOptions{
+		name: id,
+	}
+	if err := opts.validate(); err != nil {
+		return fmt.Errorf("validate alert options: %w", err)
+	}
+	sql, err := structToSQL(opts)
+	if err != nil {
+		return err
+	}
+	_, err = v.client.exec(ctx, sql)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+type ShowAlertOptions struct {
+	show   bool  `ddl:"static" sql:"SHOW"` //lint:ignore U1000 This is used in the ddl tag
+	Terse  *bool `ddl:"keyword" sql:"TERSE"`
+	alerts bool  `ddl:"static" sql:"ALERTS"` //lint:ignore U1000 This is used in the ddl tag
+
+	// optional
+	Like       *Like   `ddl:"keyword" sql:"LIKE"`
+	In         *In     `ddl:"keyword" sql:"IN"`
+	StartsWith *string `ddl:"parameter,no_equals,single_quotes" sql:"STARTS WITH"`
+	Limit      *int    `ddl:"parameter,no_equals" sql:"LIMIT"`
+}
+
+func (v *Alert) ID() SchemaObjectIdentifier {
+	return NewSchemaObjectIdentifier(v.DatabaseName, v.SchemaName, v.Name)
+}
+
+func (v *Alert) ObjectType() ObjectType {
+	return ObjectTypeAlert
+}
+
+type Alert struct {
+	CreatedOn    time.Time
+	Name         string
+	DatabaseName string
+	SchemaName   string
+	Owner        string
+	Comment      *string
+	Warehouse    string
+	Schedule     AlertSchedule
+	State        string
+	Condition    string
+	Action       string
+}
+
+type AlertSchedule interface {
+	alertSchedule()
+	String() string
+}
+
+func AlertScheduleFromString(s string) (AlertSchedule, bool) {
+	parts := strings.Fields(s)
+	if len(parts) == 2 && parts[len(parts)-1] == "MINUTE" {
+		return AlertScheduleInterval(toInt(parts[0])), true
+	} else if withoutCron := dropUsingCron(parts); len(withoutCron) == 6 {
+		return AlertScheduleCronExpression{
+			Expression: strings.Join(withoutCron[0:len(withoutCron)-1], " "),
+			TimeZone:   withoutCron[len(withoutCron)-1],
+		}, true
+	} else {
+		return nil, false
+	}
+}
+func dropUsingCron(parts []string) []string {
+	if parts[0] == "USING" && parts[1] == "CRON" {
+		return parts[2:]
+	} else {
+		return parts
+	}
+}
+
+type AlertScheduleInterval int
+
+func (interval AlertScheduleInterval) String() string {
+	return fmt.Sprintf("%d MINUTE", interval)
+}
+
+func (AlertScheduleInterval) alertSchedule() {}
+
+type AlertScheduleCronExpression struct {
+	Expression string
+	TimeZone   string
+}
+
+func (expression AlertScheduleCronExpression) String() string {
```

Comment:
the more i am looking at this the more i think it is too different than what we have been doing with other structs. Can you just use *string for schedule? And make a note to revisit this later. I do think we can have Schedule as a type, but it should not be a special implementation compared to how we build SQL statements for everything else.